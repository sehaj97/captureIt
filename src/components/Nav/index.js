import React, { useEffect } from 'react';
import { capitalizeFirstLetter } from '../../utils/helpers';

function Nav(props) {
  const {
    categories = [],
    setCurrentCategory,
    currentCategory,
  } = props;

  useEffect(() => {
    document.title = capitalizeFirstLetter(currentCategory.name);
  }, [currentCategory]);
  
  return (
    <header className="flex-row px-1">
      <h2>
        <a data-testid="link" href="/">
          <span role="img" aria-label="camera"> ðŸ“¸</span> Oh Snap!
        </a>
      </h2>
      <nav>
        <ul className="flex-row">
          <li className="mx-2">
            <a data-testid="about" href="#about">
              About me
            </a>
          </li>
          <li className="mx-2">
            <span>Contact</span>
          </li>
          {categories.map((category) => (
            <li
              className={`mx-1 ${
                currentCategory.name === category.name && 'navActive'
                }`}
              key={category.name}
            >
              <span
                onClick={() => {
                  setCurrentCategory(category)
                }}
              >
                {capitalizeFirstLetter(category.name)}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Nav;