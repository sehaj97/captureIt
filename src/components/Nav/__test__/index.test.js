import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
    //renders About test

    // First Test
  it('renders', () => {
    render(<Nav />);
  });

  it('matches snapshot DOM node structure', () => {
    // render About
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });

  });
  describe('emoji is visible', () => {
    it('inserts emoji into the h2', () => {
        render(<Nav />)
    expect(screen.getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
    });
  }) ;

  describe('links are visible', () => {
    it('inserts text into the links', () => {
      render(<Nav />);
      expect(screen.getByTestId('link')).toHaveTextContent('Oh Snap!');
      expect(screen.getByTestId('about')).toHaveTextContent('About me');
    });
  })