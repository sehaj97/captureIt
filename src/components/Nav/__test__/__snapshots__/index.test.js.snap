// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="flex-row"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
           ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
